lib/reverse_words.rb

```diff
@@ -1,4 +1,45 @@
 # A method to reverse each word in a sentence, in place.
+require 'pry'
 def reverse_words(my_words)
-  raise NotImplementedError
+  if my_words == nil || my_words.length == 0
+    return my_words
+  end
+  string_length = my_words.length
+  i = 0
+  while i < string_length
+    space = find_next_space_index(my_words, i)
+    if space == 0
+      j = 0
+    else j = space - 1
+    end
+    my_words[i..j] = string_reverse(my_words[i..j])
```

Comment:
As we saw in class, in the first few slides of https://drive.google.com/file/d/0B__DV26QHsH4bWJmS1A0QXBad1U/view?usp=sharing, the substring method e.g. `my_words[i..j]` creates a new string object. As a result, you will end up using more space and your string manipulations are not happening in-place.

Slack me if you have further questions on this.